package.json

```diff
@@ -23,29 +30,42 @@
     "@types/chai-as-promised": "^7.1.0",
     "@types/debug": "^0.0.31",
     "@types/dotenv": "^4.0.3",
+    "@types/jssha": "^2.0.0",
     "@types/mocha": "^5.2.5",
     "@types/node": "^8.0.37",
     "@types/sinon": "^5.0.5",
     "chai": "^4.2.0",
     "chai-as-promised": "^7.1.1",
+    "cross-env": "^5.2.0",
     "dotenv": "^6.1.0",
+    "events": "^3.0.0",
     "mocha": "^5.2.0",
     "mocha-junit-reporter": "^1.18.0",
     "nyc": "^12.0.2",
+    "os-browserify": "^0.3.0",
     "rhea-promise": "^0.1.7",
     "rimraf": "^2.6.2",
+    "rollup": "^1.1.2",
+    "rollup-plugin-commonjs": "^9.2.0",
+    "rollup-plugin-multi-entry": "^2.1.0",
+    "rollup-plugin-node-builtins": "^2.1.2",
+    "rollup-plugin-node-globals": "^1.4.0",
+    "rollup-plugin-node-resolve": "^4.0.0",
+    "rollup-plugin-replace": "^2.1.0",
+    "rollup-plugin-sourcemaps": "^0.4.2",
+    "rollup-plugin-uglify": "^6.0.2",
     "sinon": "^7.1.0",
     "ts-node": "^7.0.1",
     "tslint": "^5.11.0",
-    "typescript": "^3.2.2"
+    "typescript": "^3.3.3"
   },
   "scripts": {
     "tslint": "tslint -p . -c tslint.json --exclude examples/**/*.ts --exclude tests/**/*.ts",
     "tsc": "tsc",
-    "prebuild": "rimraf dist && rimraf typings",
-    "build": "npm run tslint && npm run tsc",
+    "prebuild": "rimraf dist typings dist-esm",
+    "build": "npm run tslint && npm run tsc && cross-env rollup -c",
     "test": "npm run build && npm run unit",
-    "unit": "nyc --reporter=lcov --reporter=text-lcov mocha -r ts-node/register -t 50000 ./tests/**/*.spec.ts",
+    "unit": "cross-env TS_NODE_COMPILER_OPTIONS=\"{\\\"module\\\":\\\"commonjs\\\"}\" nyc --reporter=lcov --reporter=text-lcov mocha -r ts-node/register -t 50000 ./tests/**/*.spec.ts",
```

Comment:
We could skip using `ts-node` altogether for running tests like we did in the service bus project

Reply:
Can tackle this in a separate PR if we like?